packages/renderers/renderer-lit/server.js

```diff
@@ -28,10 +36,27 @@ function* render(tagName, attrs, children) {
   const instance = new LitElementRenderer(tagName);
 
   // LitElementRenderer creates a new element instance, so copy over.
+  const Ctr = getCustomElementConstructor(tagName);
   for (let [name, value] of Object.entries(attrs)) {
-    instance.setAttribute(name, value);
+    // check if this is a property reserved by JSX parser
+    if (reservedReactProperties.has(name)) {
+      console.warn(
+        `${tagName} contains property ${name} which is a ` +
+          `reserved property. It will be used by Astro and not set on ` +
+          `the element.`
+      );
+      continue;
+    }
+    // check if this is a reactive property
+    if (name in Ctr.prototype) {
+      instance.setProperty(name, value);
+    } else {
+      instance.setAttribute(name, value);
+    }
   }
 
+  instance.connectedCallback();
```

Comment:
Should we do this @justinfagnani? I was under the impression Lit did not want connectedCallback to get called.

Reply:
Yes. This `connectedCallback` is on the ElementRenderer, so that the specific ElementRenderer implementation can decide whether to actually call the underlying element instance's `connectedCallback` or do something else. That leaves the decision up to LitElementRenderer in this case.